cloudinit/log.py

```diff
@@ -45,14 +62,20 @@ def flush_loggers(root):
     flush_loggers(root.parent)
 
 
-def define_deprecation_logger(lvl=DEPRECATED):
-    logging.addLevelName(lvl, "DEPRECATED")
+def define_extra_loggers() -> None:
+    """Add DEPRECATED and TRACE log levels to the logging module."""
+
+    def new_logger(level):
+        def log_at_level(self, message, *args, **kwargs):
+            if self.isEnabledFor(level):
+                self._log(level, message, args, **kwargs)
 
-    def deprecated(self, message, *args, **kwargs):
-        if self.isEnabledFor(lvl):
-            self._log(lvl, message, args, **kwargs)
+        return log_at_level
 
-    logging.Logger.deprecated = deprecated
+    logging.addLevelName(DEPRECATED, "DEPRECATED")
+    logging.addLevelName(TRACE, "TRACE")
+    logging.Logger.deprecated = new_logger(DEPRECATED)  # type: ignore
```

Comment:
why not `setattr` instead of `#type: ignore`?

Reply:
Done